### PR TITLE
DDS entity profiles management

### PR DESCRIPTION
**Pull request by IkerLuengo**
- Creation of entities given a profile name (instead of a QoS)
- Consider the default profile when resetting the QoS to their default value

### CODE REVIEW

File: src/cpp/fastdds/subscriber/SubscriberImpl.cpp [hunk 1]
```diff
@@ -49,6 +88,9 @@ SubscriberImpl::SubscriberImpl(
     , rtps_participant_(p->rtps_participant())
     , default_datareader_qos_(DATAREADER_QOS_DEFAULT)
 {
+    SubscriberAttributes pub_attr;
+    XMLProfileManager::getDefaultSubscriberAttributes(pub_attr);
+    set_qos_from_attributes(default_datareader_qos_, pub_attr);
```
Comment (laura-eprosima) [inline src/cpp/fastdds/subscriber/SubscriberImpl.cpp:93]:
```suggestion
    SubscriberAttributes sub_attr;
    XMLProfileManager::getDefaultSubscriberAttributes(sub_attr);
    set_qos_from_attributes(default_datareader_qos_, sub_attr);
```

File: src/cpp/fastdds/publisher/Publisher.cpp [hunk 2]
```diff
@@ -86,6 +86,15 @@ DataWriter* Publisher::create_datawriter(
     return impl_->create_datawriter(topic, qos, listener, mask);
 }
 
+DataWriter* Publisher::create_datawriter_with_profile(
+        Topic* topic,
+        const std::string& profile_name,
+        DataWriterListener* listener = nullptr,
+        const StatusMask& mask = StatusMask::all())
```
Comment (laura-eprosima) [inline src/cpp/fastdds/publisher/Publisher.cpp:93]:
```suggestion
        DataWriterListener* listener,
        const StatusMask& mask)
```

File: src/cpp/fastdds/subscriber/Subscriber.cpp [hunk 3]
```diff
@@ -80,6 +80,15 @@ DataReader* Subscriber::create_datareader(
     return impl_->create_datareader(topic, reader_qos, listener, mask);
 }
 
+DataReader* Subscriber::create_datareader_with_profile(
+        TopicDescription* topic,
+        const std::string& profile_name,
+        DataReaderListener* listener = nullptr,
+        const StatusMask& mask = StatusMask::all())
```
Comment (laura-eprosima) [inline src/cpp/fastdds/subscriber/Subscriber.cpp:87]:
```suggestion
        DataReaderListener* listener,
        const StatusMask& mask)
```

File: src/cpp/fastdds/publisher/PublisherImpl.cpp [hunk 4]
```diff
@@ -237,6 +279,24 @@ DataWriter* PublisherImpl::create_datawriter(
     return writer;
 }
 
+DataWriter* PublisherImpl::create_datawriter_with_profile(
+        Topic* topic,
+        const std::string& profile_name,
+        DataWriterListener* listener,
+        const StatusMask& mask = StatusMask::all())
```
Comment (laura-eprosima) [inline src/cpp/fastdds/publisher/PublisherImpl.cpp:286]:
```suggestion
        const StatusMask& mask)
```

File: src/cpp/fastdds/subscriber/SubscriberImpl.cpp [hunk 5]
```diff
@@ -203,6 +245,24 @@ DataReader* SubscriberImpl::create_datareader(
     return reader;
 }
 
+DataReader* SubscriberImpl::create_datareader_with_profile(
+        TopicDescription* topic,
+        const std::string& profile_name,
+        DataReaderListener* listener,
+        const StatusMask& mask = StatusMask::all())
```
Comment (laura-eprosima) [inline src/cpp/fastdds/subscriber/SubscriberImpl.cpp:252]:
```suggestion
        const StatusMask& mask)
```

File: include/fastdds/dds/domain/DomainParticipant.hpp [hunk 6]
```diff
@@ -202,6 +225,22 @@ class DomainParticipant : public Entity
...
+     * @return Pointer to the created Topic.
+     */
+    RTPS_DllAPI Topic* create_topic_with_profile(
+            const std::string& topic_name,
+            const std::string& type_name,
+            const std::string& profile_name,
+            TopicListener* listener = nullptr,
+            const StatusMask& mask = StatusMask::all());
+
     RTPS_DllAPI ReturnCode_t delete_topic(
```
Comment (MiguelCompany) [inline include/fastdds/dds/domain/DomainParticipant.hpp:244]: Could we take the chance to document this method?

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/2064/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/1820/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/2058/)

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/2069/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/1825/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/2063/)

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/2071/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/1827/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/2065/)

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/2076/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/1832/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/2070/)

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/2080/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/1836/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/2074/)

**Pull request merged**